cdap-common/src/main/java/co/cask/cdap/common/io/Locations.java

```diff
@@ -341,18 +342,22 @@ public static Location getParent(Location location) {
   }
 
   /**
-   * Returns the relative path for a given location URI
+   * Returns the relative path for a given locationURI.
+   * For failure to create URI it simply returns the Path from the given locationURI
```

Comment:
it will never fail to create a URI. The comment I had was about what happens if somebody passes in a location factory that wasn't used to create the location uri, which means they have different base paths. In that scenario, relativize just returns the input uri.